### DISCUSSION TITLE
Make a new release?

### DISCUSSION THREAD

**Discussion by catap** [beetbox/beets#4501] [Ideas]
Near a year had passed since the last release.

Maybe it isn't so bad idea to make a new one?

UPD: almost two year had passed since the last release!

**Comment by ghbrown** [comment]
Just tacking on here since I ran into this issue when trying use a feature implemented almost a year ago via the PYPI (via pip) release. I was stumped until I realized how long it had been since last release.

I ended up just getting the source and doing `pip -e .`, which was easy enough, but an actual release would be great.

**Comment by mattbarnicle** [comment]
i'd like to see a new release also. i installed beets on Void via `vpm up`, so it's running on a release cycle. there is a feature that was implemented last year (#4319) that i'd really like to have access to.

**Comment by discopatrick** [comment]
Also interested in this. It's been a while since I contributed, but I'm coming back to set up beets again on a new machine and I notice that 1.6.0 is still the latest version.

I do see however that linuxserver do nightly builds whenever we push to master here at beets - so their latest version is more up to date: https://github.com/linuxserver/docker-beets/releases

@sampsyo - hello again! And could I help with releasing a new version?

**Reply by catap** [reply]
@discopatrick if you're using Mac, you may switch to MacPorts' `beets-devel` which I'm updating to the last master each few weeks :)

**Reply by discopatrick** [reply]
Hi @catap - I do indeed have a Mac 🙂 however I tend to use Homebrew and don't use other package managers to avoid clashes.

Perhaps I could add a formula for brew though.

**Comment by catap** [comment]
@sampsyo that do you think about the new release? :) It may also announce python 3.11 as suggested and default one :)

**Comment by ghost** [comment]
I am really looking forward to setting the album_artist=${artist}, which is not supported in current stable release. 

It has been more than 1.5 years. I know release is an unpredictable thing, but are there already a certain amount of feature changes/updates that can justify a release now? @sampsyo

**Reply by benpage2** [reply]
There's no way to move album fields to track fields (and vice versa) at the moment right? I'm using mp3tag to do it currently.

**Comment by snejus** [comment]
@sampsyo @Serene-Arc @JOJ0 what are we thinking about a new release?

**Reply by JOJ0** [reply]
That everybody is too afraid and busy to do it (my personal and honest opinion without asking other project members! :-)))). BTW @snejus we need your help! I'd like to invite you to beets' internal zulip chat. I couldn't find you via beets.discourse., Please read: https://github.com/beetbox/beets/discussions/4706 and drop me a PM! Thanks!

**Reply by snejus** [reply]
Of course, will do! Would be happy to help with the release :)

**Reply by Serene-Arc** [reply]
Sounds good to me!

Though what I think we should do is recognise that there has been a problem and we should try to put in processes to make sure this doesn't happen again. Maybe we could move to weekly releases like yt-dlp, and we could put in workflows to automatically do releases.

**Reply by sampsyo** [reply]
I really like the idea of a GitHub Action to automatically release things… it would remove the temptation to be "precious" about wordsmithing the changelog, for example.

**Reply by Serene-Arc** [reply]
It would make it easier in some respects, though we'd need to change how we mark versions. An automatic workflow can't really tell what version numbers to increase AFAIK so it'd need to be a date-based system if we go down the completely automated route. Otherwise it might need a bit of a human touch.

**Reply by jpecora716** [reply]
Couldn't you configure the automatic workflow to build a new release when a new git tag is added? That way you can keep the existing versioning.

**Comment by ovizii** [comment]
I just read the changelog from November 2021 and I am wondering what happened since there are apparently no further releases since then. A bit of googling lead me here. :-)

> Changelog
> 1.6.0 (November 27, 2021)
> This release is our first experiment with time-based releases! We are aiming to publish a new release of beets every 3 months

**Reply by Serene-Arc** [reply]
Yeah...that didn't end up happening. I have implemented a release action, just waiting on the final integration.

**Comment by averms** [comment]
What is the official recommendation on running git HEAD / master in "production"?

**Reply by michaelkrieger** [reply]
See [the FAQ](https://beets.readthedocs.io/en/v1.6.0/faq.html#run-the-latest-source-version-of-beets).
Though frankly a release is very much in order.  There are some pretty big fixes in the GIT at this stage.

**Comment by catap** [comment]
I really think that https://github.com/beetbox/beets/pull/3721 worth to be released.

**Reply by Serene-Arc** [reply]
It's all almost ready, it's just taking some coordination with the team regarding some PyPi permissions.

**Reply by catap** [reply]
@Serene-Arc any news?

**Reply by Serene-Arc** [reply]
So as of two days ago or so I was added as an owner to the PyPI project so I can do the final step of releasing. So technically everything will be ready to go whenever I get to it. However since this is a completely new workflow, I want to first do a push to the testing PyPi serverrs. It's been over two years and no one wants the first release since to be mistaken in some way that will cause problems, especially since we're releasing in a completely different (and hopefully easier way).

I'm a bit busy with my studies at the moment but I'm hoping to get the testing done over the next couple of days. Hopefully you and the lovely community around beets can offer a little more patience. I know it's been a really long time but because of that, I want to be extra careful, especially since I came onto the project way after that so this will be my first time managing a beets release.

Once it's done and we know that everything works, we'll try and do more releases regularly. It'll be as simple as clicking a button and entering a tag, since I've made it as seamless as I can. That's something that the team will have to discuss because we all review PRs when we can, so there's no guarantee that doing, say, a weekly release will be worth it.

**Comment by snejus** [comment]
@Serene-Arc exciting!!!

**Comment by catap** [comment]
Just a friendly reminder that we all waiting!

**Reply by Serene-Arc** [reply]
Yup, I know. I'm in week 10 of my semester so managing to set time aside is a bit tough.

**Comment by daniele-athome** [comment]
Congrats on the new release!
I don't see relevant breaking changes for plugin developers other that this:

> beets now explicitly prevents multiple plugins to define replacement functions for the same field. When previously defining template_fields for the same field in two plugins, the last loaded plugin would silently overwrite the function defined by the other plugin. Now, beets will raise an exception when this happens.

Nothing else worth mentioning?

**Reply by Serene-Arc** [reply]
It's been a really, really long time since the last release, over two years I think. AFAIK there haven't been any other major changes to the API for plugins but I haven't been part of the project team for that entire time so I can't say for sure.

**Comment by catap** [comment]
And 2.0.0 were released, so, let me close this discussion as completed!
